docs/Daedalus-Installer-Update-Procedure.md

```diff
@@ -0,0 +1,267 @@
+# Daedalus Installer Update Procedure
+
+After the installers are built in CI, they need to be copied to the
+installer updates S3 bucket, and an update proposal needs to be
+submitted to the cardano-sl network.
+
+This is how wallet clients will self-update.
+
+**New!** There have recently been some changes and improvements to
+these commands from [DEVOPS-868][].
+
+## Running the scripts
+
+`ssh` into the deployer get a `nix-shell` in the `iohk-ops` checkout.
+
+    cd mainnet
+    nix-shell -A withAuxx
+
+You can see the list of subcommands with:
+
+    io update-proposal --help
+
+Each `update-proposal` subcommand has its own usage information
+accessible with the `--help` option.
+
+| **Subcommand**      | **Synopsis**                                        |
+| ------------------- | --------------------------------------------------- |
+| init                | Create template config file and working directory.  |
+| find-installers     | Download installer files from the Daedalus build.   |
+| sign-installers     | Sign downloaded installer files with GPG.           |
+| upload-s3           | Upload installer files to the S3 bucket.            |
+| set-version-json    | Update the version info file in the the S3 bucket.  |
+| submit              | Send update proposal transaction to the network.    |
+
+The commands are run in order with manual checks between each step.
+
+## Requirements
+
+1. The keys for core nodes should be in the `keys` subdirectory.
+
+2. You need to know the following things:
+
+   * Daedalus revision to propose.
+   * The `lastKnownBlockVersion`, according to the table below.
+   * The `voterIndex`, according to the table below.
+   * Signing key passphrase (optional).
+   * IP address of a privileged relay (use `io info`).
+
+## 1. Initialise params
+
+    io -c NETWORK.yaml update-proposal init [DATE] --revision REVISION --block-version VER --voter-index N
+
+Where *DATE* is a string which identifies the update proposal. By
+default it will be today's date in `YYYY-MM-DD` format. The other
+values should be set as required.
+
+This command will create a template config file like
+`update-proposals/mainnet-2018-04-03/params.yaml`.
+
+The directory that the config file is created in is called the
+*work dir*. It will contain logs, a node db, keys, installers, hashes,
+and other information generated by the update proposal.
+
+Open `params.yaml` in the work dir and double-check that it has
+correct values. The contents will look similar to this:
+
+    voterIndex: 2
+    daedalusRevision: 32daff63e1fb1590cd7320e4253e61b2a47b0963
+    lastKnownBlockVersion: '0.1.0'
+
+
+## 2. Download installer files from CI
+
+    io -c NETWORK.yaml update-proposal find-installers DATE
+
+The *DATE* parameter should be the same as what was used for the
+`init` step.
```

Comment:
What if `DATE` was not provided but inferred? Does does it infer the current date here as well?